core/store/migrations/migration1586163842/migrate.go

```diff
@@ -0,0 +1,65 @@
+package migration1586163842
+
+import (
+	"github.com/jinzhu/gorm"
+)
+
+// Migrate adds foreign keys that were missing
+func Migrate(tx *gorm.DB) error {
+	// Add a few more useful indexes while we are here. This also speeds up
+	// some queries in this migration
+	err := tx.Exec(`
+	CREATE INDEX idx_task_specs_job_spec_id ON task_specs (job_spec_id);
+	CREATE INDEX idx_job_runs_run_request_id ON job_runs (run_request_id);
+	CREATE INDEX idx_job_runs_result_id ON job_runs (result_id);
+	CREATE INDEX idx_job_runs_initiator_id ON job_runs (initiator_id);
+	CREATE INDEX idx_task_runs_result_id ON task_runs (result_id);
+	`).Error
+	if err != nil {
+		return err
+	}
+
+	// Need to cast initiators.job_spec_id to UUID to be compatible with referenced key
+	err = tx.Exec(`
+	ALTER TABLE initiators ALTER COLUMN job_spec_id TYPE uuid USING job_spec_id::uuid;
+	`).Error
+	if err != nil {
+		return err
+	}
+
+	// Before we were using 0 as the null value, need to set to explicit null for FK
+	err = tx.Exec(`
+	UPDATE job_runs SET result_id = NULL WHERE result_id = 0;
+	UPDATE job_runs SET run_request_id = NULL WHERE run_request_id = 0;
+	UPDATE job_runs SET initiator_id = NULL WHERE initiator_id = 0;
+	UPDATE task_runs SET result_id = NULL WHERE result_id = 0;
+	UPDATE task_runs SET task_spec_id = NULL WHERE task_spec_id = 0;
+	`).Error
+	if err != nil {
+		return err
+	}
+
+	// This was assumed by the code and probably would have crashed/hung if this assumption was violated.
+	// Let's make that explicit.
+	err = tx.Exec(`
+	ALTER TABLE job_runs ALTER COLUMN initiator_id SET NOT NULL;
```

Comment:
Just above, you run
```
UPDATE job_runs SET initiator_id = NULL WHERE initiator_id = 0;
```
...and then here, you make that column `NOT NULL`.  What does Postgres do if there are now `NULL` values for some rows?

Reply:
Fail with error.

Good catch, we should probably just delete any job runs without an initiator_id.

Reply:
I'll add this to the release notes so that node operators are aware.